### PR TITLE
Removal of GET and POST usage

### LINKED ISSUE #2: Is it practical to have the GET version altogether?
All I've found so far on why there's both the GET and the POST version of DoH in HTTP is that GETs are cachable.

In CoAP, we have [FETCH](https://datatracker.ietf.org/doc/html/rfc8132), which is just there for cases like this, and does away with both contender's downsides:
* Compared to POST, it's cachable (and safe and thus idempotent)
* Compared to GET, it doesn't need
  * URI template handling (which I'd discourage in an embedded context -- it's reasonably simple here, but I wouldn't want to have to think of the different rules that'd apply if the template occurs in the path or query string), and
  * base64 encoding (which adds both processing and wire overhead).

The only upside I see of keeping GET/POST over FETCH is that DoC could then be implemented by just cross-proxying to a DoH server, but given that such a proxy is unconstrained, reasonably straightforward to implement on its own and can thus take load off the constrained client, I advocate going all FETCH.

### PR DESCRIPTION
**Pull request by miri64**
Resolves #2.

### CODE REVIEW

File: draft-lenders-dns-over-coap.md [hunk 1]
```diff
@@ -353,6 +353,40 @@ Reference: [TBD-this-spec]
 
 --- back
 
+Other CoAP Request Methods
+==========================
+This document only introduces the usage of the FETCH method for CoAP requests
+carrying DNS queries. If for whatever reason other methods are required to be
+used, request methods that can carry payloads, such as POST or PUT, are
```
Comment (chrysn) [inline draft-lenders-dns-over-coap.md:360]: I'm fine with this appendix, but why is PUT introduced? Unlike POST which has intentionally vague semantics, PUT has defined semantics ("that the resource [...] be updated or created with the enclosed representation") that don't match what's being done here.

**Review by chrysn — APPROVED** [review]

File: draft-lenders-dns-over-coap.md (hunk 1 above)
Comment (waehlisch) [reply draft-lenders-dns-over-coap.md:360]: i agree that this approach could be a good compromise. but this also means that we abandon the C-H-Proxy argument right away.

Comment (miri64) [reply draft-lenders-dns-over-coap.md:360]: > I'm fine with this appendix, but why is PUT introduced?

I guess I was in "people might do anything if they don't get propper instructions"-mode here. I can remove it again, if it is more distracting than useful.

> but this also means that we abandon the C-H-Proxy argument right away.

I was thinking about adding a sentence or two about how C-H-Proxies could be implemented in §5. Of course, for this recommendations should be amended here as well then.

Comment (waehlisch) [reply draft-lenders-dns-over-coap.md:360]: we cannot propose  C-H-Proxies in the body of the draft and specify the required building blocks in the Appendix. if we move GET to the Appendix, the proxy option is out-of-scope for the draft.

Comment (miri64) [reply draft-lenders-dns-over-coap.md:360]: > we cannot propose C-H-Proxies in the body of the draft and specify the required building blocks in the Appendix.

That was not what I intended. The building blocks specific to the main part of the draft would of course be specified in the main part of the draft. Recommendations specific to the appendix (such as how to handle GET/POST at a C-H-Proxy) would go to the appendix.

**Comment by miri64** [comment]
And another rebase. If we agree with the current state, I'd like to have this merged soon (and maybe even publish `-01` soon after).

**Review by chrysn — APPROVED** [review]

**Comment by waehlisch** [comment]
> And another rebase. If we agree with the current state, I'd like to have this merged soon (and maybe even publish `-01` soon after).

as i said earlier. i would like to have a discussion in the CoRE WG before we change this. this discussion can be kicked off after your vacation @miri64.

**Pull request merged**